### PR TITLE
Add UnmarshalBSON MarshalBSONValue to bson template

### PR DESCRIPTION
**Pull request by dadrichem**

### CODE REVIEW

**Comment by olistrik** [comment]
@Daan-Adrichem did we want to be committing the vendor folder? it wasnt there before.

File: examples/named/named_day_enum_marshal_bson.go [hunk 1]
```diff
@@ -3,6 +3,9 @@ package day
 
 import (
 	"github.com/globalsign/mgo/bson"
+
+	mongo "go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
```
Comment (olistrik) [inline examples/named/named_day_enum_marshal_bson.go:8]: were now requiring users to add both drivers to their dependencies? Feels like this should be an option;

```
-bson=[ mgo | mongo-driver ]
```

**Review by olistrik — COMMENTED** [review]
@dadrichem reviewed

File: examples/named/named_day_enum_marshal_bson.go (hunk 1 above)
Comment (dadrichem) [reply examples/named/named_day_enum_marshal_bson.go:8]: and if you want both?

Comment (dadrichem) [reply examples/named/named_day_enum_marshal_bson.go:8]: I build this support because I want both, I dont see a reason not to have both either

**Comment by dadrichem** [comment]
> @Daan-Adrichem did we want to be committing the vendor folder? it wasnt there before.

The git ignore does not specify therefor it should be included IMO. but which ever approach you think is appropriate

**Comment by olistrik** [comment]
> > @Daan-Adrichem did we want to be committing the vendor folder? it wasnt there before.
> 
> The git ignore does not specify therefor it should be included IMO. but which ever approach you think is appropriate

Huh? The gitignore is controlled by us. Vendor isnt in there not because we decided to commit vendor when I set the project up, but because we never thought about it and Ive never run `go vendor` in this directory.

File: examples/named/named_day_enum_marshal_bson.go (hunk 1 above)
Comment (olistrik) [reply examples/named/named_day_enum_marshal_bson.go:8]: Why would you need both database drivers? Do you have two connections to your DB using different drivers?

Comment (dadrichem) [reply examples/named/named_day_enum_marshal_bson.go:8]: I do, I am migrating slowly from 1 to the other driver

**Comment by dadrichem** [comment]
> .

@olistrik I am happy to not commit the vendor folder but then it should go in the git ignore

**Comment by jerbob92** [comment]
The options are not:
 - Have vendor in the gitignore or
 - Always commit/push a vendor folder

You can also just not push it, the gitignore just protects you from accidentally pushing it. I would say that libraries should never have the vendor folder included.

**Comment by dadrichem** [comment]
> The options are not:
> 
> * Have vendor in the gitignore or
> * Always commit/push a vendor folder
> 
> You can also just not push it, the gitignore just protects you from accidentally pushing it. I would say that libraries should never have the vendor folder included.

@jerbob92 then ill add it to the git ignore and not commit the vendor folder

**Pull request merged**